docs/architecture/psa-migration/md-cipher-dispatch.md

```diff
@@ -477,3 +503,57 @@ The architecture can be extended to support `MBEDTLS_PSA_CRYPTO_CLIENT` with a l
 
 * Compile-time dependencies: instead of checking `defined(MBEDTLS_PSA_CRYPTO_C)`, check `defined(MBEDTLS_PSA_CRYPTO_C) || defined(MBEDTLS_PSA_CRYPTO_CLIENT)`.
 * Implementers of `MBEDTLS_PSA_CRYPTO_CLIENT` will need to provide `psa_can_do_hash()` (or a more general function `psa_can_do`) alongside `psa_crypto_init()`. Note that at this point, it will become a public interface, hence we won't be able to change it at a whim.
+
+### Cipher light
+
+#### Definition
+
+**Note:** this definition is tentative an may be refined when implementing and
+testing, based and what's needed by internal users of Cipher light. The new
+config symbol will not be considered public so its definition may change.
+
+Cipher light will be automatically enabled in `build_info.h` by modules that
+need it, namely: CTR\_DRBG, CCM, GCM. Note: CCM and GCM currently depend on
+the full `CIPHER_C` (enforced by `check_config.h`); this hard dependency would
+be replaced by the above auto-enablement.
+
+Cipher light includes:
+- info functions;
+- support for block ciphers in ECB mode, encrypt only (note: in Cipher, "ECB"
+  means just one block, contrary to PSA);
+- the one-shot API as well as (part of) the streaming API;
+- only AES, Aria and Camellia.
+
+This excludes:
+- the AEAD/KW API (both one-shot and streaming);
+- support for stream ciphers;
+- support for other modes of block ciphers (CBC, CTR, CFB, etc.);
+- DES and variants (3DES).
+
+The following API functions, and supporting types, are candidates for
+inclusion in the Cipher light API, with limited features as above:
+```
```

Comment:
Our starting point should be the list of functions called by the modules we want to support:

```
$ grep -hEo 'mbedtls_cipher_[A-Z_a-z0-9]+' library/ccm.c library/gcm.c library/psa_crypto_aead.c | sort -u
mbedtls_cipher_free
mbedtls_cipher_id_t
mbedtls_cipher_info_from_psa
mbedtls_cipher_info_from_values
mbedtls_cipher_info_get_block_size
mbedtls_cipher_info_t
mbedtls_cipher_setkey
mbedtls_cipher_setup
mbedtls_cipher_update
```

The modules don't actually call `mbedtls_cipher_init` (they just set everything to 0, which is what `mbedtls_cipher_init` does internally), but we should include it for completeness and to be able to test cipher-light (I expect `test_suite_cipher` test cases for ECB to be enabled with cipher-light).

`mbedtls_cipher_info_from_psa` is currently needed to make GCM and CCM work through PSA when they aren't directly implemented by a driver. Is not in `cipher.c` (it's currently in `psa_crypto_cipher.c`). It relies on `mbedtls_cipher_info_from_values`.

(I wrote the comment above before f1878d89741ed968b55c2a10ce5d996073bd74bf and I see you've now reached the same list as me. I'm still posting this comment because I think this document should explain how we came up with that list.)

Reply:
Side point: I didn't know `grep -ho`, I had always been using `sed -n s/.*\(pattern\).*/\1/p` to achieve the same result. Thanks for the tip!